src/index.js

```diff
@@ -0,0 +1,144 @@
+// Literacy
+// ========
+// Literate programming in JavaScript using reStructuredText.
+//
+//
+// Register ``.js.rst``` File Extension
+// ------------------------------------
+// Intercept Node module loading for filenames ending in ``.js.rst``.  This
+// allows ``.js.rst`` files to be passed to ``require`` calls without including
+// the suffix.
+//
+// Implementation is copied from `register.coffee`_ in CoffeeScript and is similar
+// to `babel-register`_.
+//
+// .. _register.coffee: https://github.com/jashkenas/coffeescript/tree/master/src/register.coffee
+// .. _babel-register: https://github.com/babel/babel/tree/master/packages/babel-register
+//
+// The implementation uses the ``require.extensions`` global in Node. Setting a
+// key in this dictionary adds Node handling for new file extensions. The key is
+// the file suffix string to support, the value is a function to call to load and
+// process the file.
+//
+// But ``require.extenions`` has been deprecated in Node because there is the
+// potential for conflicts when a suffix is used by two modules simultaneously.
+// It also encourages the offloading of compilation to runtime instead of build
+// time.
+//
+// Strictly speaking it should not be used. However:
+//
+// - Only this module uses ``.js.rst`` suffixes.
+// - The Node module containing the ``require.extensions`` implementation is
+//   locked so Node is committed to keeping the current form indefinitely.
+// - Production use cases are encouraged to precompile ``.js.rst`` sources.
+//
+// For ``.js.rst`` loads, we currently only read the file contents and strip the
+// Unicode byte order mark if it is present. This will be extended in following
+// work to extract JavaScript codeblocks from the RestructuredText, preform the
+// compilation and include the output into the running environment.
+//
+// .. code-block:: javascript
+
+    const fs = require('fs')
+    const stripBom = require('strip-bom')
+
+    const loadFile = function(module, filename) {
+      var content = fs.readFileSync(filename, 'utf8')
+      content = stripBom(content)
+
+      console.log(content)
```

Comment:
So I reckon this is temporary?

Reply:
Yes, #2 extracts the JavaScript blocks from `content` and passed them to `Module._compile` instead of this.